### PR TITLE
Gremlin 3.6.1

### PR DESCRIPTION
**Pull request by fabrice404**
- Updates gremlin to 3.6.1, and other dependencies to latest versions.
- Removes deprecated `connectOnStartup`, but still connects when instantiated.
- Adds pre-commit hooks for linting and testing

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [12.x, 14.x, 16.x]
```
Comment (JoyfulShush) [inline .github/workflows/node.js.yml:19]: This is a breaking change - you should upgrade your package version to the next major version in this case and clearly document it.

Is it necessary to drop the Node 12 support?

If possible, maybe create two new versions - one that simply supports gremlin 3.6.1 and fixes it (patch) and one that upgrades all other dependencies while dropping Node 12 (major)?

The major upgrade is more important for vulnerability fixes, but if someone is using Node 12 they're essentially stuck until they upgrade.

File: test/unit/driver/aws-sigv4-driver-remote-connection.test.js [hunk 2]
```diff
@@ -204,11 +204,8 @@ describe('AwsSigV4DriverRemoteConnection', () => {
 
   describe('open', () => {
     it('should open the client connection', () => {
-      const connection = new AwsSigV4DriverRemoteConnection(HOST, PORT, OPTS);
-      const open = jest.fn();
-      connection._client = { open };
+      const connection = new AwsSigV4DriverRemoteConnection(HOST, PORT, { ...OPTS, openOnStartup: false }, () => { expect(1).toEqual(1); });
```
Comment (JoyfulShush) [inline test/unit/driver/aws-sigv4-driver-remote-connection.test.js]: I don't think this test is running correctly. By the time you get a callback, your test is already done. I'm not familiar with jest, but chai has `chaiAsPromised` which lets you handle async operations such as this.

You should wrap this in a promise and make the promise be resolved in the callback.

**Review by JoyfulShush — CHANGES_REQUESTED** [review]

File: test/unit/driver/aws-sigv4-driver-remote-connection.test.js (hunk 2 above)
Comment (fabrice404) [reply test/unit/driver/aws-sigv4-driver-remote-connection.test.js]: Good catch, I've just put that `expect(1).toEqual(1)` temporarily and forgot to write it properly.

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (fabrice404) [reply .github/workflows/node.js.yml:19]: Node 12 has reached its EOL on April 2022, the package will still work with it, but the tests will not. https://nodejs.org/en/about/releases/

People concerned by vulnerabilities should really update their node version.

**Review by JoyfulShush — APPROVED** [review]

**Review by andygnewman — APPROVED** [review]
👍

**Pull request merged**